libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -445,11 +445,16 @@ string YulUtilFunctions::typedShiftLeftFunction(Type const& _type, Type const& _
 			Whiskers(R"(
 			function <functionName>(value, bits) -> result {
 				bits := <cleanAmount>(bits)
-				result := <cleanup>(<shift>(bits, value))
+				<?leftAligned>
+					result := <shift>(bits, <cleanup>(value))
+				<!leftAligned>
+					result := <cleanup>(<shift>(bits, value))
+				</leftAligned>
```

Comment:
In `typedShiftRightFunction` we do `cleanup(shift(bits, cleanup(value))` unconditionally. Do you mind briefly explaining what is different between left and right shifts that we need a conditional `<?leftAligned>` here but not in right shift function.

To put it differently, what if we always do `result := <cleanup>(<shift>(bits, <cleanup>(value))`?

Reply:
I just ran some tests and it turns out that (at least for unsigned numbers), the optimizer is clever enough to remove one of the cleanups, so we can also simplify this to the same case as the right shift.